scripts/3.r

```diff
@@ -0,0 +1,151 @@
+# Assignment 1:  
+library(tweedie) 
+library(ggplot2)
+library(doParallel)
+
+simTweedieTest <-  
+  function(N){ 
+    t.test( 
+      rtweedie(N, mu=10000, phi=100, power=1.9), 
+      mu=10000 
+    )$p.value 
+  } 
+
+
+# Assignment 2: 
+maxcores <- 8
+Cores <- min(parallel::detectCores(), maxcores)
+
+# Instantiate the cores:
+cl <- makeCluster(Cores)
+
+# Next we register the cluster
+registerDoParallel(cl)
+
+# Trying to fix an error: Error in { : task 1 failed - "could not find function "simTweedieTest""
+clusterExport(cl, c("simTweedieTest", "rtweedie", "t.test"))
+
+
+MTweedieTests <-  
+  function(N,M,sig){ 
+    result <- foreach(i = 1:M, .combine ="c")%dopar%{
+      simTweedieTest(N) < sig
+    }
+    share_reject <- sum(result)/M
+    return(share_reject)
+  } 
```

Comment:
Function looks good, but you should have included lines 15-26 and 56 inside the function. This is to ensure that you stop the parallization for each iteration in the data frame.